drivers/sensor/adxl372/adxl372.c

```diff
@@ -0,0 +1,1104 @@
+/*
+ * Copyright (c) 2018 Analog Devices Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <string.h>
+#include <sensor.h>
+#include <init.h>
+#include <gpio.h>
+#include <misc/printk.h>
+#include <misc/__assert.h>
+#include <stdlib.h>
+#include <spi.h>
+#include <i2c.h>
+
+#include "adxl372.h"
+
+static int adxl372_bus_access(struct device *dev, u8_t reg,
+			      void *data, size_t length)
+{
+	struct adxl372_data *adxl372_data = dev->driver_data;
+
+#ifdef CONFIG_ADXL372_SPI
+	const struct spi_buf buf[2] = {
+		{
+			.buf = &reg,
+			.len = 1
+		}, {
+			.buf = data,
+			.len = length
+		}
+	};
+
+	struct spi_buf_set tx = {
+		.buffers = buf,
+	};
+
+	if (reg & ADXL372_READ) {
+		const struct spi_buf_set rx = {
+			.buffers = buf,
+			.count = 2
+		};
+
+		tx.count = 1;
+
+		return spi_transceive(adxl372_data->bus,
+				      &adxl372_data->spi_cfg, &tx, &rx);
+	}
+
+	tx.count = 2;
+
+	return spi_write(adxl372_data->bus, &adxl372_data->spi_cfg, &tx);
+#elif CONFIG_ADXL372_I2C
+	const struct adxl372_dev_config *cfg = dev->config->config_info;
+
+	if (reg & ADXL372_READ) {
+		return i2c_burst_read(adxl372_data->bus, cfg->i2c_addr,
+				      ADXL372_TO_I2C_REG(reg),
+				      (u8_t *) data, length);
+	} else {
+		return i2c_reg_write_byte(adxl372_data->bus, cfg->i2c_addr,
+					  ADXL372_TO_I2C_REG(reg),
+					  *(u8_t *)data);
```

Comment:
This looks better with `i2c_reg_write_byte` but I still don't understand why in this particular case you assume `length` is 1. In every other case (spi read/write, i2c read), you use the passed value.

Reply:
write is slow path - we only do burst reads.
So for write - the upper layer code uses always length=1.
Does it make sense to use i2c_burst_write_addr(), knowingly we just add overhead.
Or shall I return error if length!=1? Or leave as is?